dipy/reconst/qtdmri.py

```diff
@@ -0,0 +1,2118 @@
+                                                       tau[bval_mask])
+                tau_scaling = ut / us.mean()
+                tau_scaled = tau * tau_scaling
+                ut /= tau_scaling
+                us = np.clip(us, self.eigenvalue_threshold, np.inf)
+                q = np.dot(bvecs, R) * qvals[:, None]
+                M = qtdmri_signal_matrix_(
+                    self.radial_order, self.time_order, us, ut, q, tau_scaled,
+                    self.normalization
+                )
+            else:
+                us, ut = qtdmri_isotropic_scaling(data_norm, qvals, tau)
+                tau_scaling = ut / us
+                tau_scaled = tau * tau_scaling
+                ut /= tau_scaling
+                R = np.eye(3)
+                us = np.tile(us, 3)
+                q = bvecs * qvals[:, None]
+                M = qtdmri_signal_matrix_(
+                    self.radial_order, self.time_order, us, ut, q, tau_scaled,
+                    self.normalization
+                )
+        else:
+            us, ut = qtdmri_isotropic_scaling(data_norm, qvals, tau)
+            tau_scaling = ut / us
+            tau_scaled = tau * tau_scaling
+            ut /= tau_scaling
+            R = np.eye(3)
+            us = np.tile(us, 3)
+            q = bvecs * qvals[:, None]
+            M = qtdmri_isotropic_signal_matrix_(
+                self.radial_order, self.time_order, us[0], ut, q, tau_scaled,
+                normalization=self.normalization
+            )
+
+        b0_indices = np.arange(self.gtab.tau.shape[0])[self.gtab.b0s_mask]
+        tau0_ordered = self.gtab.tau[b0_indices]
+        unique_taus = np.unique(self.gtab.tau)
+        first_tau_pos = []
+        for unique_tau in unique_taus:
+            first_tau_pos.append(np.where(tau0_ordered == unique_tau)[0][0])
+        M0 = M[b0_indices[first_tau_pos]]
+
+        lopt = 0.
+        alpha = 0.
+        if self.laplacian_regularization and not self.l1_regularization:
+            if self.cartesian:
+                laplacian_matrix = qtdmri_laplacian_reg_matrix(
+                    self.ind_mat, us, ut, self.S_mat, self.T_mat, self.U_mat,
+                    self.part1_reg_mat_tau,
+                    self.part23_reg_mat_tau,
+                    self.part4_reg_mat_tau,
+                    normalization=self.normalization
+                )
+            else:
+                laplacian_matrix = qtdmri_isotropic_laplacian_reg_matrix(
+                    self.ind_mat, us, ut, self.part1_uq_iso_precomp,
+                    self.part1_reg_mat_tau, self.part23_reg_mat_tau,
+                    self.part4_reg_mat_tau,
+                    normalization=self.normalization
+                )
+            if self.laplacian_weighting == 'GCV':
+                try:
+                    lopt = generalized_crossvalidation(data_norm, M,
+                                                       laplacian_matrix)
+                except:  # noqa: E722
```

Comment:
Can you remove `# noqa: E722`and replace it by `except BaseException:`

Can you do it for all of them?

Reply:
Changed.